packages/react-core/src/modules/in-app-message/hooks/useInAppMessages.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  FetchInAppMessagesOptions,
+  InAppMessage,
+  InAppMessageClient,
+} from "@knocklabs/client";
+import { useStore } from "@tanstack/react-store";
+import { useEffect, useMemo } from "react";
+
+import { useInAppMessageChannel } from "../context";
+
+// TODO: Type so that the message content (or fields) is typed
+export const useInAppMessages = (
+  messageType: string,
+  options: FetchInAppMessagesOptions = {},
+) => {
+  const { inAppChannelClient } = useInAppMessageChannel();
+
+  const inAppMessageClient = useMemo(() => {
+    return new InAppMessageClient(inAppChannelClient, messageType, options);
+  }, [inAppChannelClient, messageType, options]);
+
+  // TODO: Is it possible to extract all store usage?
+  // Maybe not because framework specific hooks need to be a called to support reactivity
+  // Could definitely create selectors as functions in the client library though
+  const messages = useStore(inAppChannelClient.store, (state) => {
+    const messageIds = new Set(
+      state.queries[inAppMessageClient.queryKey]?.data?.items?.map(
+        (message) => message.id,
+      ),
+    );
+
+    const messages = Object.entries(state.messages).reduce<InAppMessage[]>(
+      (messages, [id, message]) => {
+        if (messageIds.has(id)) {
+          messages.push(message);
+        }
+        return messages;
+      },
+      [],
+    );
+
+    return messages;
+  });
+
+  const { networkStatus, loading } = useStore(
```

Comment:
I'm curious why we have two separate useStore() calls here and above? Asking because both are using the same inAppChannelClient.store.

Reply:
Yeah might be nice to have a single `useStore` call that retrieves the query and operates on that?

Reply:
Yeah, I'll consolidate these and make some other updates! I think I'm going to update the `query` object to only store the message ids as well based on some other feedback